src/js/src/views/SolrWayback.vue

```diff
@@ -41,55 +43,61 @@ export default {
       solrSettings: state => state.Search.solrSettings
     }),
   },
+
   mounted() {
     window.addEventListener('scroll', this.onScroll)
   },
   methods: {
     ...mapActions('Search', {
-      requestSearch: 'requestSearch',
-      requestFacets: 'requestFacets',
-      updateQuery: 'updateQuery',
-      requestImageSearch: 'requestImageSearch',
-      requestUrlSearch: 'requestUrlSearch',
-      updateSearchAppliedFacets:'updateSearchAppliedFacets',
+      resetState:'resetState',
+      addToSearchAppliedFacets:'addToSearchAppliedFacets',
+      emptySearchAppliedFacets:'emptySearchAppliedFacets',
       updateSolrSettingGrouping:'updateSolrSettingGrouping',
       updateSolrSettingImgSearch:'updateSolrSettingImgSearch',
       updateSolrSettingUrlSearch:'updateSolrSettingUrlSearch',
-      updateFutureSolrSettingGrouping:'updateFutureSolrSettingGrouping',
-      updateFutureSolrSettingImgSearch:'updateFutureSolrSettingImgSearch',
-      updateFutureSolrSettingUrlSearch:'updateFutureSolrSettingUrlSearch'
+      updateSolrSettingOffset:'updateSolrSettingOffset',
     }),
     onScroll(e) {
     e.target.documentElement.scrollTop > 0 ? this.scrolledFromTop = true : this.scrolledFromTop = false
     },
     backToTop() {
       window.scrollTo({ top: 0, behavior: 'smooth' })
-    }, 
+    },
   },
   beforeRouteUpdate (to, from, next) {
-    //console.log('route changed!',to)
-    this.updateQuery(to.query.q)
-    to.query.grouping === 'true' ? (this.updateSolrSettingGrouping(true), this.updateFiutureSolrSettingGrouping(true)) : (this.updateSolrSettingGrouping(false), this.updateFutureSolrSettingGrouping(false))
-    to.query.imgSearch === 'true' ? (this.updateSolrSettingImgSearch(true), this.updateFutureSolrSettingImgSearch(true)) : (this.updateSolrSettingImgSearch(false), this.updateFutureSolrSettingImgSearch(false))
-    to.query.urlSearch === 'true' ? (this.updateSolrSettingUrlSearch(true), this.updateFutureSolrSettingUrlSearch(true)) :  (this.updateSolrSettingUrlSearch(false), this.updateFutureSolrSettingUrlSearch(false))
-    to.query.facets ? this.updateSearchAppliedFacets(to.query.facets) : this.updateSearchAppliedFacets('')
-    if(this.solrSettings.imgSearch) {
-      this.requestImageSearch({query:this.query})
+    //console.log('route changed!',to.query, from.query)
+    // Check if any of our params have changed. Could be refactored into a nice functon.
+    if(to.query.query !== undefined &&
+       (to.query.query !== from.query.query ||
+       to.query.offset !== from.query.offset ||
+       to.query.imgSearch !== from.query.imgSerach ||
+       to.query.urlSearch !== from.query.urlSearch ||
+       to.query.grouping !== from.query.grouping ||
+       to.query.facets !== from.query.facets)) {
```

Comment:
Se my comment about a similar block of code in HistoryRoutingUtils

Reply:
I've made it into a function instead, in this case.